featurebyte/models/entity_lookup_feature_table.py

```diff
@@ -0,0 +1,244 @@
+    return out
+
+
+def _get_entity_lookup_graph(
+    lookup_step: EntityLookupStep,
+    feature_store: FeatureStoreModel,
+) -> EntityLookupGraphResult:
+    relation_table = lookup_step.relation_table
+    graph = QueryGraph()
+    input_node = graph.add_operation_node(
+        node=relation_table.construct_input_node(feature_store_details=feature_store),
+        input_nodes=[],
+    )
+
+    child_column_name = None
+    parent_column_name = None
+    feature_dtype = None
+    for column_info in relation_table.columns_info:
+        if column_info.entity_id == lookup_step.child_entity.id:
+            child_column_name = column_info.name
+        elif column_info.entity_id == lookup_step.parent_entity.id:
+            parent_column_name = column_info.name
+            feature_dtype = column_info.dtype
+    assert child_column_name is not None
+    assert parent_column_name is not None
+    assert feature_dtype is not None
+
+    additional_params: Dict[str, Any]
+    if relation_table.type == TableDataType.SCD_TABLE:
+        assert isinstance(relation_table, SCDTableModel)
+        additional_params = {
+            "scd_parameters": SCDBaseParameters(
+                effective_timestamp_column=relation_table.effective_timestamp_column,
+                natural_key_column=relation_table.natural_key_column,
+                current_flag_column=relation_table.current_flag_column,
+                end_timestamp_column=relation_table.end_timestamp_column,
+            )
+        }
+    elif relation_table.type == TableDataType.EVENT_TABLE:
+        assert isinstance(relation_table, EventTableModel)
+        additional_params = {
+            "event_parameters": {
+                "event_timestamp_column": relation_table.event_timestamp_column,
+            }
+        }
+    else:
+        # TODO: handle ITEM_TABLE which also needs event_parameters
+        additional_params = {}
+    lookup_node = graph.add_operation(
+        node_type=NodeType.LOOKUP,
+        node_params={
+            "input_column_names": [parent_column_name],
+            "feature_names": [lookup_step.parent_entity.serving_names[0]],
+            "entity_column": child_column_name,
+            "serving_name": lookup_step.child_entity.serving_names[0],
+            "entity_id": lookup_step.child_entity.id,
+            **additional_params,
+        },
+        node_output_type=NodeOutputType.FRAME,
+        input_nodes=[input_node],
+    )
+    feature_node = graph.add_operation(
+        node_type=NodeType.PROJECT,
+        node_params={"columns": [lookup_step.parent_entity.serving_names[0]]},
+        node_output_type=NodeOutputType.SERIES,
+        input_nodes=[lookup_node],
+    )
+    feature_job_setting_extractor = FeatureJobSettingExtractor(graph=graph)
+    feature_job_setting = (
+        feature_job_setting_extractor.extract_from_agg_node(node=lookup_node)
+        or feature_job_setting_extractor.default_feature_job_setting
```

Comment:
I realized this shouldn't try to default to `default_feature_job_setting` as it should indeed be None in some cases. Will fix in a follow up.

Reply:
Could we use `extract` method here?

Reply:
Yes, that's what I meant. Should just use `extract_from_agg_node` and remove `or feature_job_setting_extractor.default_feature_job_setting`.